group_project_v2/group_project.py

```diff
@@ -594,7 +612,62 @@ def dashboard_detail_view(self, context):
 
         return fragment
 
-    def _build_groups_data(self, workgroups, stage_stats):
+    def __render_user(self, user, stage_stats, filtered_users):
+        """
+        :param group_project_v2.project_api.dtos.ReducedUserDetail user:
+
+        :return: dict
+        """
+
+        return {
+            'id': user.id, 'full_name': user.full_name, 'email': user.email,
+            'is_filtered_out': user.id in filtered_users,
+            'stage_states': {
+                stage_id: stage_data.get('user_stats', {}).get(user.id, StageState.UNKNOWN)
+                for stage_id, stage_data in stage_stats.iteritems()
+            },
+            'groups_to_grade': {
+                stage_id: [
+                    {'id': group.id, 'ta_grade_link': self.get_ta_review_link(group.id, stage_id)}
+                    for group in stage_data.get('groups_to_grade', {}).get(user.id, [])
+                ]
+                for stage_id, stage_data in stage_stats.iteritems()
+            }
+        }
+
+    def __render_workgroup(self, workgroup, stage_stats, children_context):
+        """
+
+        :param group_project_v2.project_api.dtos.WorkgroupDetails workgroup:
+        :return: dict
+        """
+
+        users = [
+            self.__render_user(user, stage_stats, children_context)
+            for user in workgroup.users
+        ]
+
+        users.sort(key=lambda u: u['is_filtered_out'])
+
+        group_visible = any((not u['is_filtered_out'] for u in users))
```

Comment:
@jbzdak nit: `u` is not very good name, even for loop variable. I wonder why pylint have not yelled at you for this :)